### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by Mo7amedSMahdi**
# Create a database based on a diagram

### Project requirements
- database diagram for a clinic:
     -   
- Created a file named `schema_based_on_diagram.sql` where we implement the database from the diagram.
- Join tables from many-to-many relationships might not appear in the diagram, but you still need them.
- Remember to add the FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+);
+
+CREATE TABLE medical_treatment (
+    id SERIAL PRIMARY KEY,
+    medical_id INT,
+    treatment_id INT,
+    CONSTRAINT fk_medical_history FOREIGN KEY("medical_id") REFERENCES "public"."medical_histories"("id") ON DELETE NO ACTION ON UPDATE NO ACTION,
+    CONSTRAINT fk_medical_treatment FOREIGN KEY("treatment_id") REFERENCES "public"."treatments"("id") ON DELETE NO ACTION ON UPDATE NO ACTION
+);
+
```
Comment (gemmen29) [inline schema_based_on_diagram.sql:48]: - [x] Kindly create a table to represent the relation between `medical_histories` and `treatment` as the relationship is many to many. don't forget to create indexes for the foreign keys in that table.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯  
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    CONSTRAINT treatment_id FOREIGN KEY ("treatment_id") REFERENCES "public"."treatments" ("id") ON DELETE NO ACTION ON UPDATE NO ACTION
+);
+
+CREATE TABLE medical_treatment (
+    id SERIAL PRIMARY KEY,
+    medical_id INT,
+    treatment_id INT,
+    CONSTRAINT fk_medical_history FOREIGN KEY("medical_id") REFERENCES "public"."medical_histories"("id") ON DELETE NO ACTION ON UPDATE NO ACTION,
+    CONSTRAINT fk_medical_treatment FOREIGN KEY("treatment_id") REFERENCES "public"."treatments"("id") ON DELETE NO ACTION ON UPDATE NO ACTION
+);
```
Comment (Mo7amedSMahdi) [inline schema_based_on_diagram.sql:47]: Hi @gemmen29 as you can see I already created the table for the medical_history and treatments
Kindly check if there are any issues with this table so we can solve them.

Comment (gemmen29) [reply schema_based_on_diagram.sql:47]: Ok, sorry I didn't notice it just create the indexes for those foreign keys.

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (gemmen29) [reply schema_based_on_diagram.sql:48]: The table was implemented you just need to implement the indexes of that table(medical_treatment).

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+-- Optimized Execution analysis by creating Indexes
+CREATE INDEX treatment_id_asc ON invoice_items (treatment_id ASC);
+CREATE INDEX invoices_asc ON invoice_items (invoice_id ASC);
+CREATE INDEX patients_asc ON medical_histories (patient_id ASC);
+CREATE INDEX medical_histories_id_asc ON invoices (medical_histories_id ASC);
```
Comment (Mo7amedSMahdi) [inline schema_based_on_diagram.sql]: @gemmen29 And here is the indexes

Comment (gemmen29) [reply schema_based_on_diagram.sql]: @Mo7amedSMahdi The indexes for the `medical_treatment` table had two foreign keys is not implemented.

**Review by mateo951 — APPROVED** [review]
### Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💣
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**